aspnetcore/fundamentals/configuration.md

```diff
@@ -5,162 +5,981 @@ ms.author: riande
+
+````none
+
+   using Microsoft.AspNetCore.Mvc;
+   using Microsoft.Extensions.Options;
+
+   public class HomeController : Controller
+   {
+       private readonly IOptions<MyWindow> _optionsAccessor;
+
+       public HomeController(IOptions<MyWindow> optionsAccessor)
+       {
+           _optionsAccessor = optionsAccessor;
+       }
+       public IActionResult Index()
+       {
+           var height = _optionsAccessor.Value.Height;
+           var width = _optionsAccessor.Value.Width;
+           var left = _optionsAccessor.Value.Left;
+           var top = _optionsAccessor.Value.Top;
+
+           return Content($"height = {height}, width = {width}, "
+                        + $"Left = {left}, Top = {top}");
+       }
+   }
+   ````
+
+
+  ### GetValue
+
+The following sample demonstrates the `GetValue<T>` extension method:
+
+<!-- literal_block {"xml:space": "preserve", "language": "none", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "source": "/Users/shirhatti/src/Docs/aspnet/fundamentals/configuration/sample/src/InMemoryGetValue/Program.cs", "highlight_args": {"linenostart": 1, "hl_lines": [25, 26, 27, 28, 29]}, "names": []} -->
+
+````none
+
+   using Microsoft.Extensions.Configuration;
+   using System;
+   using System.Collections.Generic;
+
+   // Add NuGet  <package id="Microsoft.Extensions.Configuration.Binder"
+   public class Program
+   {   
+       static public IConfigurationRoot Configuration { get; set; }
+       public static void Main(string[] args = null)
+       {
+           var dict = new Dictionary<string, string>
+               {
+                   {"Profile:MachineName", "Rick"},
+                   {"App:MainWindow:Height", "11"},
+                   {"App:MainWindow:Width", "11"},
+                   {"App:MainWindow:Top", "11"},
+                   {"App:MainWindow:Left", "11"}
+               };
+
+           var builder = new ConfigurationBuilder();
+           builder.AddInMemoryCollection(dict);
+           Configuration = builder.Build();
+           Console.WriteLine($"Hello {Configuration["Profile:MachineName"]}");
+
+           // Show GetValue overload and set the default value to 80
+           // You typically would set default values in the constructor.
+           // Requires NuGet package "Microsoft.Extensions.Configuration.Binder"
+           var left = Configuration.GetValue<int>("App:MainWindow:Left", 80);
+           Console.WriteLine($"Left {left}");
+
+           var window = new MyWindow();
+           Configuration.GetSection("App:MainWindow").Bind(window);
+           Console.WriteLine($"Left {window.Left}");
+       }
+   }
+   ````
+
+The ConfigurationBinder’s `GetValue<T>`  method allows you to specify a default value (80 in the sample). Default values are more easily set in the class constructor. `GetValue` is basically just syntax sugar for `GetSection(<key>).TypeConverter.Convert<T>`, it's not doing any binding.
```

Comment:
> Default values are more easily set in the class constructor.

Not sure this remark is relevant. `GetValue<T>()` does not bind to entire sections, just gets scalar values form `GetSection(key).Value` converted to a specific type directly without doing binding in simple scenarios, hence getting a default value can be handy. 

> `GetValue` is basically just syntax sugar for `GetSection(<key>).TypeConverter.Convert<T>`,

Not sure what this pseudo code means. `GetValue<T>` can be considered shortcut syntax for something like `TypeDescriptor.GetConverter(typeof<T>).ConvertFromInvariantString(configuration.GetSection(key).Value)`, but not sure pseudo code helps.